src/lib/idmap/sss_idmap.exports

```diff
@@ -34,7 +34,7 @@ SSS_IDMAP_0.4 {
         sss_idmap_free_smb_sid;
         sss_idmap_free_bin_sid;
         idmap_error_string;
-        is_domain_sid;
+        is_str_sid;
```

Comment:
> Hi,
> 
> it is not recommended to change the API of a released library. Please move `is_domain_sid()` and `is_principal_sid()` to `sss_idmap.c` and add `is_principal_sid()` and if you like `is_str_sid()` to a new library version `SSS_IDMAP_0.5`.
> 
> bye, Sumit

Can't we just move (for backward compatibility) the `is_domain_sid()` function? I mean `is_principal_sid()` can stay as inline in `sss_idmap.h` right?

Reply:
> > Hi,
> > it is not recommended to change the API of a released library. Please move `is_domain_sid()` and `is_principal_sid()` to `sss_idmap.c` and add `is_principal_sid()` and if you like `is_str_sid()` to a new library version `SSS_IDMAP_0.5`.
> > bye, Sumit
> 
> Can't we just move (for backward compatibility) the `is_domain_sid()` function? I mean `is_principal_sid()` can stay as inline in `sss_idmap.h` right?

Hi,

I have to admit that do not like code in header files that much. I know there are a few examples in the SSSD code. But libsss_idmap is a library used by other projects as well and I would prefer to make it clear that a call part of the API or not. So I would like to encourage you to add `is_principal_sid()` to `sss_idmap.c` as well. If you think that `is_principal_sid()` does not belong into libsss_idmap I would suggest to just add it to `sdap_async_nested_groups.c` as a macro calling `is_str_sid()`.

bye,
Sumit

Reply:
Hi,

thank you, this looks good now. I just got mixed up with the version numbers `SSS_IDMAP_0.5` is the current one. So can you please mofify the change of `sss_idmap.exports` to
```
--- a/src/lib/idmap/sss_idmap.exports
+++ b/src/lib/idmap/sss_idmap.exports
@@ -64,3 +64,12 @@ SSS_IDMAP_0.5 {
         sss_idmap_add_auto_domain_ex;
 
 } SSS_IDMAP_0.4;
+
+SSS_IDMAP_0.6 {
+
+    # public functions
+    global:
+
+        is_principal_sid;
+
+} SSS_IDMAP_0.5;
```

Finally I would like to suggest to add `fspdn_len` to `group_ctx->opts->sdom` as well so that `strlen(fspdn)` has to be calculated only once.

bye,
Sumit